File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size();
        int majorityElement = nums[0];
        int count = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] == majorityElement) {
                count++;
            } else {
                count--;
            }
            if (count == 0) {
                majorityElement = nums[i];
                count++;
            }
        }
        return majorityElement; // Need not check coz given that it always exists
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length;
        int majority = nums[0];
        int counter = 1;
        for (int i = 1; i < n; i++) {
            if (nums[i] == majority) {
                counter += 1;
            } else {
                counter -= 1;
            }
            if (counter == 0) {
                majority = nums[i];
                counter += 1;
            }
        }
        return majority;
    }
}